crates/solver/src/solver/naive_solver/multi_order_solver.rs

```diff
@@ -279,14 +279,26 @@ fn compute_uniswap_in(
 /// Thus we ensure that `buy_token_price / sell_token_price >= limit_buy_amount / limit_sell_amount`
 ///
 fn is_valid_solution(solution: &Settlement) -> bool {
-    for order in solution.traded_orders() {
+    for (order, is_liquidity_order) in solution
+        .traded_user_orders()
+        .map(|order| (order, false))
+        .chain(
+            solution
+                .traded_liquidity_orders()
+                .map(|order| (order, true)),
+        )
+    {
         let order = &order.creation;
-        let buy_token_price = solution
-            .clearing_price(order.buy_token)
-            .expect("Solution should contain clearing price for buy token");
         let sell_token_price = solution
             .clearing_price(order.sell_token)
             .expect("Solution should contain clearing price for sell token");
+        let buy_token_price = if is_liquidity_order {
+            sell_token_price / order.sell_amount * order.buy_amount
```

Comment:
Is it maybe * sell_amount/ buy_amount?


Let's assume that the order is selling 10*18 wei for 1 USDC = 3000* 10**6 USDC wei.

Further we assume the prices are nominated, such that 1 ETH = 10**18.
Then buy price should be according to your formula:
buy_token_price(USDC) = 10**18/(10**18) * 3000 * 10**6 =3*10**9.

But usually usdc prices are higher than 10**18. 
https://logs.gnosis.io/app/discover#/doc/de1f5360-a9ae-11ea-acbb-9d2e3cb124ca/logstash-2022.04.13?id=YkOKIoABkQPmW2Y8R4g3

Reply:
I think you are right. I checked a few more examples, with regular 10e18 base tokens and the formula is wrong. Was blindly following the description on the top of the function.
Fixed. Thanks.